pkgs/distributions/default.nix

```diff
@@ -0,0 +1,107 @@
+{
+  fetchFromGitHub,
+  callPackage,
+  fetchPypi,
+  eigen,
+  parsable,
+  goftests,
+  protobuf3_20,
+  python3Packages,
+}:
+let
+  version = "2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "posterior";
+    repo = "distributions";
+    rev = "43c11618b0f229682fb916612ba2437c5f22a753"; # there is no tag
+    sha256 = "sha256-DiJ6Ljwc5K1CrzzexAQ53g86sKqaroYRhmXuxAHAOq4=";
+  };
+
+  distributions-shared = callPackage ./distributions-shared.nix { inherit version src; };
+
+  imageio = python3Packages.buildPythonPackage rec {
+    pname = "imageio";
+    version = "2.6.1";
+
+    src = fetchPypi {
+      inherit pname version;
+      hash = "sha256-9E6yMbnfSFh08v/SLf0MPHEefeB2UWuTdO3qXGW8Z64=";
+    };
+
+    doCheck = false;
+
+    nativeBuildInputs = with python3Packages; [
+      pytest
+    ];
+
+    propagatedBuildInputs = with python3Packages; [
+      pillow
+    ];
+
+    buildInputs = with python3Packages; [
+      enum34
+      numpy
+    ];
+  };
+in
+python3Packages.buildPythonPackage {
+  pname = "distributions";
+
+  inherit version src;
+
+  nativeBuildInputs = [
+    protobuf3_20
+    python3Packages.pyflakes
+  ];
+
+  buildInputs = [
+    eigen
+    # TODO: we're not sure if this is even needed
+    distributions-shared
+    protobuf3_20
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    protobuf3_20
+    protobuf
+    cython
+    numpy
+    parsable
+    scipy
+    simplejson
+  ];
+
+  # TODO: be more precise. Some tests seem to be still in Python 2.
+  doCheck = false;
+  nativeCheckInputs = with python3Packages; [
+    imageio
+    nose
+    goftests
+    pytest
+  ];
+
+  preBuild = ''
+    make protobuf
+  '';
+
+  patches = [
+    ./use-imread-instead-of-scipy.patch
```

Comment:
@Schaechtle used here is a source code patch ...

Reply:
cc @srounce , do you have any links to github issues or other info that would contextualize how we would know this adaptation is no longer needed?

Reply:
This is needed due to the version of Scipy being newer than that expected by the tests. `imread` has been deprecated from Scipy for a while it seems, and ImageIO provides a drop-in replacement. This can be removed when distributions updates the test to use `imread` from ImageIO.

https://imageio.readthedocs.io/en/stable/user_guide/scipy.html